doc/wcmp/wcmp-design.md

```diff
@@ -0,0 +1,605 @@
+2. show: display switch WCMP global configuration
+
+### 1.2.3 Error handling
+
+#### 1.2.3.1 Frontend
+
+**This feature will provide error handling for the next situations:**
+1. Missing parameter value
+2. Invalid parameter value
+
+#### 1.2.3.2 Backend
+
+**This feature will provide error handling for the next situations:**
+1. Invalid parameter value
+2. Parameter removal
+3. Configuration removal
+
+### 1.2.4 Event logging
+
+#### 1.2.4.1 Frontend
+
+**This feature will provide event logging for the next situations:**
+1. Configuration update
+
+###### Table 1: Frontend event logging
+
+| Event                         | Severity |
+|:------------------------------|:---------|
+| Configuration update: success | NOTICE   |
+| Configuration update: error   | ERROR    |
+
+#### 1.2.4.2 Backend
+
+**This feature will provide event logging for the next situations:**
+1. Invalid parameter value
+2. Parameter removal
+3. Configuration removal
+4. Configuration update
+
+###### Table 2: Backend event logging
+
+| Event                         | Severity |
+|:------------------------------|:---------|
+| Invalid parameter value       | ERROR    |
+| Parameter removal             | ERROR    |
+| Configuration removal         | ERROR    |
+| Configuration update: success | NOTICE   |
+| Configuration update: error   | ERROR    |
+
+# 2 Design
+
+## 2.1 Overview
+
+![WCMP design](images/wcmp_design.svg "Figure 1: WCMP design")
+
+###### Figure 1: WCMP design
+
+A network fabric is a type of network topology where all nodes, in this case switches and endpoints,  
+are interconnected to all other nodes. In such a topology, full bisection bandwidth is often expected.  
+The full bisection bandwidth allows one half of the network nodes to communicate simultaneously  
+with the other half of the nodes.
+
+When a link fails, it is important not only for routing topology to converge quickly, but also converge  
+in a way that can maintain the best performance possible. For example, with multiple links connecting  
+each ToR and Spine, one link failing does not change reachability, but has big implications  
+in terms of performance due to reduced capacity.
+
+Weighted-Equal Cost-Multipath (W-ECMP) offers a solution to this problem in which the relative link weight  
+reflecting the available capacity is distributed using the BGP bandwidth community attribute.
+
+Dataplane configuration flow:
+* The BGP link bandwidth extended community carries information about the available link capacity for the route  
+prefixes through the network, which gets mapped to the weight of the corresponding next hop. The mapping  
+factors the bandwidth value of a particular path against the total bandwidth values of all possible paths,  
+normalized to the range 1 to 255
+* FRR converts the incoming BGP link bandwidth extended community values into proportionated weight  
+among the ecmp members in such a way that the cumulative value of individual weights is normalized to 255
```

Comment:
AS individual weight is normalized to be between 1 and 255, what does  "the cumulative value of individual weights  is normalized to 255" mean?

Reply:
@philo-micas the answer to this question will be added later to the official documentation here: [weighted-ecmp-using-bgp-link-bandwidt](https://docs.frrouting.org/en/latest/bgp.html#weighted-ecmp-using-bgp-link-bandwidth)
The relevant request was forwarded to the FRR team.